src/passes/SafeHeap.cpp

```diff
@@ -117,12 +122,21 @@ struct SafeHeap : public Pass {
     // add imports
     addImports(module);
     // instrument loads and stores
-    AccessInstrumenter(getSbrkPtr).run(runner, module);
+    std::set<Name> ignoreFunctions;
+    if (module->start.is()) {
+      FindCalledFunctions findCalledFunctions;
+      findCalledFunctions.walkFunctionInModule(
+        module->getFunction(module->start), module);
```

Comment:
Doesn't this need to compute the "closure" here, that is if A calls B which calls C then we want C in this set as well?

Taking a step back, though, maybe another approach is to do this at runtime. That seems safer than statically ignoring a potentially large chunk of the program. What I mean here is that if the issue is that the sbrk location has not yet been set then we can just early exit from SAFE_HEAP functions. Then they will start to check the sbrk location only after it is set to a valid (>0) value?

Reply:
Actually for my use case its enough to only avoid instrumenting the direct callees because the format of the start function is relatively fixed.  Its either:

a. `__wasm_memory_init` itself
b. A function that directly calls `__wasm_memory_init`

Its never more complex than that today.   So this change fixes the current issue.  If its OK with you we can land this as is, and we can look into a smarter/runtime solution separately?

Reply:
BTW the start function can never call user-generated code, only very specific linker-generated code, so the risk of this "infecting" other part of the program don't exist.

Reply:
Good points, I guess we don't need to handle anything more generic then. Please add a comment though explaining that.

Reply:
Done.